src/iden3comm/handlers/contract-request.ts

```diff
@@ -193,14 +199,17 @@ export class ContractRequestHandler
    */
   private async createContractInvokeResponse(
     request: ContractInvokeRequest,
-    txHashToZkpResponseMap: Map<string, ZeroKnowledgeProofResponse[]>
+    txHashToZkpResponseMap: Map<string, ZeroKnowledgeProofResponse[]>,
+    ctx: ContractMessageHandlerOptions
   ): Promise<ContractInvokeResponse> {
+    const recipient = getIden3CommSingleRecipient(request);
+    const target = request.from && ctx.multipleRecipientsFormat ? [request.from] : request.from;
     const contractInvokeResponse: ContractInvokeResponse = {
       id: request.id,
       thid: request.thid,
       type: PROTOCOL_MESSAGE_TYPE.CONTRACT_INVOKE_RESPONSE_MESSAGE_TYPE,
-      from: request.to,
-      to: request.from,
+      from: recipient ? recipient.string() : undefined,
+      to: target,
```

Comment:
maybe it's make sense to create helper function `createBasicMsg(..params)` 
that hides logic of multiple `to` passing `thid`, `mediaTyp` etc. 
And when you creates concrete message you could do something like that
```msg = {
    ...createBasicMsg(thid, to, isMultipleRecipients..., typ),
body: {...}
type: ide3MsgType
}
```